weblate/fonts/views.py

```diff
@@ -83,23 +86,67 @@ def post(self, request: AuthenticatedHttpRequest, **kwargs):
 @method_decorator(login_required, name="dispatch")
 class FontDetailView(ProjectViewMixin, DetailView):
     model = Font
+    _font_form = None
 
     def get_queryset(self):
         return self.project.font_set.all()
 
     def get_context_data(self, **kwargs):
         result = super().get_context_data(**kwargs)
         result["can_edit"] = self.request.user.has_perm("project.edit", self.project)
+        result["font_form"] = self._font_form or FontForm()
         return result
 
     def post(self, request: AuthenticatedHttpRequest, **kwargs):
         self.object = self.get_object()
+
         if not request.user.has_perm("project.edit", self.project):
             raise PermissionDenied
 
-        self.object.delete()
-        messages.error(request, gettext("Font deleted."))
-        return redirect("fonts", project=self.project.slug)
+        if "font" not in request.FILES:
```

Comment:
This really should use form validation instead of doing all the checks manually.